services/services.go

```diff
@@ -1698,6 +1707,72 @@ func latestExportedStatisticDayUpdater(wg *sync.WaitGroup) {
 	}
 }
 
+func latestTotalSupplyUpdater(wg *sync.WaitGroup) {
+	firstRun := true
+	for {
+		genesisTotalSupply := utils.Config.Chain.GenesisTotalSupply
+
+		totalAmountWithdrawn, _, err := db.GetTotalAmountWithdrawn()
+		if err != nil {
+			logger.Errorf("error getting total amount withdrawn data: %v", err)
+			time.Sleep(time.Second * 30)
+			continue
+		}
+
+		latestData := LatestState()
+		if err != nil {
+			logger.Errorf("error getting latest state data: %v", err)
+			time.Sleep(time.Second * 30)
+			continue
+		}
+
+		chainIDBig := new(big.Int).SetUint64(utils.Config.Chain.Config.DepositChainID)
+		rpcClient, err := rpc.NewLighthouseClient("http://"+utils.Config.Indexer.Node.Host+":"+utils.Config.Indexer.Node.Port, chainIDBig)
+		if err != nil {
+			logger.Errorf("error creating new rpc CL client instnace: %v", err)
+			time.Sleep(time.Second * 30)
+			continue
+		}
+
+		// Get the total staked gwei that was active (i.e., able to vote) during the latestFinalizedEpoch epoch
+		validatorParticipation, err := rpcClient.GetValidatorParticipation(latestData.CurrentFinalizedEpoch)
+		if err != nil {
+			logger.Errorf("error getting GetValidatorParticipation data: %v", err)
+			time.Sleep(time.Second * 30)
+			continue
+		}
+
+		latestBurnData := LatestBurnData()
+		address := common.FromHex(strings.TrimPrefix(utils.Config.Chain.Config.DepositContractAddress, "0x"))
+
+		addressMetadata, err := db.BigtableClient.GetMetadataForAddress(address)
+		if err != nil {
+			logger.Errorf("error getting GetMetadataForAddress data: %v", err)
+			time.Sleep(time.Second * 30)
+			continue
+		}
+
+		depositContractBalanceGWei := decimal.NewFromBigInt(new(big.Int).SetBytes(addressMetadata.EthBalance.Balance), 0).DivRound(decimal.NewFromInt(params.GWei), 18)
+
+		// Deposit contract holds 320k LYX lost forever, so we reduce by 320000000000000 GWei
+		totalSupply := (genesisTotalSupply + totalAmountWithdrawn + validatorParticipation.EligibleEther) - (uint64(depositContractBalanceGWei.InexactFloat64()) + uint64(latestBurnData.TotalBurned) + uint64(320000000000000))
+		amount := new(big.Int).Mul(new(big.Int).SetUint64(totalSupply), big.NewInt(params.GWei))
+
+		cacheKey := fmt.Sprintf("%d:frontend:latestTotalSupply", utils.Config.Chain.Config.DepositChainID)
+		err = cache.TieredCache.SetString(cacheKey, amount.String(), time.Hour*24)
+		if err != nil {
+			logger.Errorf("error caching last total supply: %v", err)
+		}
+		if firstRun {
+			firstRun = false
+			wg.Done()
+			logger.Info("initialized  last total supply updater")
+		}
+		ReportStatus("latestTotalSupply", "Running", nil)
+		time.Sleep(time.Second * 768)
```

Comment:
Magic number? Why 768 specifically?

Reply:
Epoch = 32 slots
Slot = 12s
I want to wait 2 epochs (because of finalization in Gasper - Casper FFG). Then we can consider it's final and we can refresh `totalSupply` imo.